packages/grid_client/src/clients/tf-grid/contracts.ts

```diff
@@ -107,28 +165,30 @@ class TFContracts extends Contracts {
           }`;
     try {
       const response = await gqlClient.query(body, { contractId: options.id });
-      const billReports = response["data"]["contractBillReports"];
+      const gqlConsumption: GqlConsumption = response["data"] as GqlConsumption;
+      const billReports = gqlConsumption.billReports;
       if (billReports.length === 0) {
         return 0;
       } else {
         let duration: number;
-        const amountBilled = new Decimal(billReports[0]["amountBilled"]);
+        const amountBilled = new Decimal(billReports[0].amountBilled);
         if (billReports.length === 2) {
-          duration = (billReports[0]["timestamp"] - billReports[1]["timestamp"]) / 3600; // one hour
+          duration = (billReports[0].timestamp - billReports[1].timestamp) / 3600; // one hour
         } else {
-          const nodeContracts = response["data"]["nodeContracts"];
-          const nameContracts = response["data"]["nameContracts"];
-          const rentContracts = response["data"]["rentContracts"];
           let createdAt: number;
-          for (const contracts of [nodeContracts, nameContracts, rentContracts]) {
+          for (const contracts of [
+            gqlConsumption.contracts.nodeContracts,
+            gqlConsumption.contracts.nameContracts,
+            gqlConsumption.contracts.rentContracts,
+          ]) {
             if (contracts.length === 1) {
-              createdAt = contracts[0]["createdAt"];
-              duration = (billReports[0]["timestamp"] - createdAt) / 3600;
+              createdAt = +contracts[0].createdAt;
+              duration = (billReports[0].timestamp - createdAt) / 3600;
               break;
             }
           }
         }
-        if (!duration) {
+        if (!duration!) {
```

Comment:
Why do we need the second exclamation mark?
 if duration is null, undefined, or any other falsy value, it will be set to 1.